js/animation/position.js

```diff
@@ -293,26 +294,50 @@ const calculatePosition = function(what, options) {
 
         let boundaryWidth = windowWidth;
         let boundaryHeight = windowHeight;
+        let containerWidth = windowWidth;
+        let containerHeight = windowHeight;
 
         if(boundary) {
             const $boundary = $(boundary);
             const boundaryPosition = $boundary.offset();
 
-            left = boundaryPosition.left;
-            top = boundaryPosition.top;
+            boundaryLeft = boundaryPosition.left;
+            boundaryTop = boundaryPosition.top;
 
             boundaryWidth = $boundary.width();
             boundaryHeight = $boundary.height();
         }
 
+        let horizontalMin = boundaryLeft + h.boundaryOffset;
+        let horizontalMax = boundaryLeft + boundaryWidth / hZoomLevel - h.mySize - h.boundaryOffset;
+        let verticalMin = boundaryTop + v.boundaryOffset;
+        let verticalMax = boundaryTop + boundaryHeight / vZoomLevel - v.mySize - v.boundaryOffset;
+
+        if(container && !isWindow(container)) {
+            const $container = $(container);
```

Comment:
looks like we can just update `top` and `left` here this way:
```
top = Math.min(top, containerPosition.top);
top = Math.min(left, containerPosition.left);
```

and not to change any calculations

Reply:
it can be used for 'min' bound calculation, but for 'max' we should use containerWidth/Height. And it is unnecessary after ['adjustBoundary'](https://github.com/DevExpress/DevExtreme/pull/15545#discussion_r536046578) function extraction